### ISSUE TITLE
get_url on a Github https url cant validate certificate

### ISSUE THREAD

**Issue by peerst** [ansible/ansible-modules-core#1210]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.7.2
##### Environment:

Mac OSX 10.10.3 (host) FreeBSD 10.1 (guest) via Vagrant
##### Summary:

get_url on a Github https url cant validate certificate
##### Steps To Reproduce:

```
- name: Install ca_root_nss
  pkgng: name=security/ca_root_nss
- name: Ensure /opt/local/bin
  file: path=/opt/local/bin state=directory mode=0755
- name: Download kerl
  get_url: url="https://raw.githubusercontent.com/spawngrid/kerl/master/kerl" dest=/opt/local/bin mode=0755
```
##### Expected Results:

It should work and validate certs: FreeBSDs "fetch" command and "curl" download this ok and and validate certs by default.
##### Actual Results:

```
TASK: [Install ca_root_nss] *************************************************** 
ok: [default]

TASK: [Ensure /opt/local/bin] ************************************************* 
ok: [default]

TASK: [Download kerl] ********************************************************* 
failed: [default] => {"dest": "/opt/local/bin", "failed": true, "gid": 0, "group": "wheel", "mode": "0755",     "owner": "root", "response": "Request failed: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED]   certificate verify failed (_ssl.c:581)>", "size": 512, "state": "directory", "status_code": -1, "uid": 0, "url":  "https://raw.githubusercontent.com/spawngrid/kerl/master/kerl"}
msg: Request failed

FATAL: all hosts have already failed -- aborting
```

**Comment by peerst** [comment]
Tried the same with validate_certs "no" and get the same error:

```
- name: Download kerl
  get_url: url="https://raw.githubusercontent.com/spawngrid/kerl/master/kerl" dest=/opt/local/bin mode=0755 validate_certs=no
```

**Comment by bcoca** [comment]
I just tested with 1.9.0.1 and it works, we will not be patching 1.7.x for this so I suggest upgrading. There were several fixes and improvements relating to SSL in 1.8 and 1.9.

**Comment by peerst** [comment]
Have upgraded to 1.9.0.1 and it works for me too

Thanks!
